### ISSUE TITLE
Keras model import issue

### ISSUE THREAD

**Issue by montardon** [eclipse/deeplearning4j#7798]
#### Issue Description

I load a Keras model stored in h5 file. It is a ResNet20 to be run on Cifar10.

I have to manually set learning rates to get scores that are not NaN.
It still have problem afterwards with my model not predicting all classes.

The Keras model runs fine.

#### Version Information

* Deeplearning4j version: 1.0.4 beta
* platform information (OS, etc): Ubuntu 18.04
* CUDA version, if used : 10.0
* NVIDIA driver version, if in use: 418.67

#### Contributing

I can help.

[keras_import.tar.gz](https://github.com/deeplearning4j/deeplearning4j/files/3228036/keras_import.tar.gz)

**Comment by maxpumperla** [comment]
@montardon we need some more info from you. so you want to train in dl4j? not sure why you would have to set the learning rate to get predictions otherwise. also, which classes "work" and which don't? can you give me a stack trace (pref. as gist) that shows what doesn't work and how you try to use it?

**Comment by montardon** [comment]
@maxpumperla I want to train the model I have, stored in .h5. 
It was written using Keras and I trained it with Keras. 
Now, I want to do the same with DL4J. I reload the model and want to train it with DL4J.
Basically, I am testing how difficult it is to switch from Keras/Python to DL4J/Java.
By working, I mean the Keras model trains well in Keras/Python environment.

**Comment by maxpumperla** [comment]
@montardon oh, I see. So you don't get good training results for the imported model? But the import itself works.

Ok, in that case just the model isn't enough. Most of the time when we get issues like these, the problem tends to be on the data preprocessing side in Java. Can you show your Python and Java code each?

**Comment by montardon** [comment]
Well, I was not sure so I filled an issue. But I'm certainly doing something wrong.
https://gist.github.com/montardon/e22b4b0d5e111c76a78d73cabf530bdc is the Java part.
https://gist.github.com/montardon/9d8740f704e9d1ed3b075e9493025109 for Keras model
https://gist.github.com/montardon/ee20ae6b66aa4813d4a92ddb8dab14fc for training output in DL4J

**Comment by maxpumperla** [comment]
@montardon you're right to file an issue, that's what they're there for. 

Looks like you mapped lr schedule etc. correctly.  don't see anything wrong.

Just curious, have you tried setting higher learning rates in dl4j? I think a while back @wmeddie noticed such a behavior as well. Also, I'm curious to see if other learning-related parameters are correctly mapped on import, since you have to set the learning rate - we don't support import of custom schedules yet.

**Comment by AlexDBlack** [comment]
> Just curious, have you tried setting higher learning rates in dl4j? I think a while back @wmeddie noticed such a behavior as well.

@maxpumperla The issue you are thinking of was fixed many releases back.

**Comment by montardon** [comment]
My only concern now is the training, not been made as in Keras.

Setting a learning rate at 0.001 fixed NaN score after first iteration.
My input images are now "normalized" (values between -0.5 and 0.5).

So I'm expecting results quite similar to Keras training.
And it looks like only a few classes are predicted...
What should I check ?

**Comment by agibsonccc** [comment]
Closing, not relevant anymore.
